crates/edr_provider/src/data.rs

```diff
@@ -362,9 +362,16 @@ impl ProviderData {
             let code = state
                 .basic(address)?
                 .map_or(Ok(Bytes::new()), |account_info| {
-                    state
-                        .code_by_hash(account_info.code_hash)
-                        .map(|bytecode| bytecode.bytecode)
+                    state.code_by_hash(account_info.code_hash).map(|bytecode| {
```

Comment:
This should be implemented by the `Bytecode::original_bytes` function already

Reply:
Thanks, I missed that somehow. Fixed in [0ddb3c5](https://github.com/NomicFoundation/hardhat/pull/4703/commits/0ddb3c56b51c6237df8396e343c7c742b219b0f1)

(Note that both `code_by_hash` and `original_bytes` clone so we end up cloning twice instead of moving the `Bytes` out of the `Bytecode` returned by `code_by_hash`, but I don't think it matters as this is not a hot path, so I think the cleaner implementation with `original_bytes` is better.)